docs_src/advanced/column_types/tutorial003.py

```diff
@@ -0,0 +1,73 @@
+from datetime import UTC, datetime
+from pathlib import Path
+from uuid import UUID, uuid4
+
+from pydantic import EmailStr, IPvAnyAddress
+from sqlmodel import Field, Session, SQLModel, create_engine, select
+
+
+class Avatar(SQLModel, table=True):
+    id: UUID = Field(default_factory=uuid4, primary_key=True)
+    source_ip_address: IPvAnyAddress
+    upload_location: Path
```

Comment:
I was surprised to find sqlmodel/pydantic don't convert these values to string for persistence or back into the IP/Path types on retrieval. I think the solution is to implement a Pydantic type, but perhaps I'm doing something wrong here?

Reply:
I think this is a bug in the pydantic v2 implementation because it used to convert it back to native python types. I've seen it happen with datetimes as well.